src/openforms/utils/json_logic.py

```diff
@@ -49,3 +49,57 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
         return JsonLogicTest.from_expression(value)
 
     raise NotImplementedError(f"Unknown value type: {type(value)}")
+
+def find_component_key(test: JsonLogicTest, candidates: list) -> str:
+    for value in test.values:
+        if isinstance(value, JsonLogicTest) and value.operator == "var":
+            candidate = value.values[0]
+            if candidate in candidates:
+                return candidate
+
+        elif isinstance(value, JsonLogicTest) and value.operator == "date":
+            return find_component_key(value, candidates=candidates)
+
+    return ""
+
+
+def find_key_value(value: Union[JsonLogicTest, str]) -> str :
+    """Find the value associated to the key in the json_trigger_rule"""
+    if isinstance(value, JsonLogicTest) :
+        return find_key_value(value.values[0])
+
+    return value
```

Comment:
I'm not sure what this does? What happens if the order of `values` is swapped?

Reply:
@sergei-maertens It retrieves the value compared in the condition no matter we compare either the field to a value, or another field/date :
![Capture d’écran du 2022-06-30 15-10-52](https://user-images.githubusercontent.com/61249514/176685685-31236d45-d89c-4ff8-887d-2a57076a3c26.png)